### PR TITLE
Closes #8679: Emit top sites count fact for top sites telemetry

### LINKED ISSUE #8679: Emit a top sites count Fact for Top Sites Telemetry
In https://github.com/mozilla-mobile/fenix/issues/15265, we want to be able to emit the top sites size count when we call `getTopSites` in [DefaultTopSitesStorage](https://github.com/mozilla-mobile/android-components/blob/be232773113848eb2ebd270b17271448423aee46/components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt#L89) so that we can report whether or not the user has top sites for telemetry.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by gabrielluong** [comment]
Gonna wait for codecov to be happy 🗡️

**Comment by rocketsroger** [comment]
> Gonna wait for codecov to be happy 🗡️

I don't think the codecov result is accurate.  Except TopSitesFacts.kt (which is straight forward call to Fact so I didn't add a test).  Everything else that the codecov commented on are tested.  -9.47% doesn't look right.

**Comment by rocketsroger** [comment]
Added more test but the coverage is worst 😱

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt [hunk 1]
```diff
@@ -20,6 +29,10 @@ class TopSitesUseCases(topSitesStorage: TopSitesStorage) {
          */
         operator fun invoke(title: String, url: String, isDefault: Boolean = false) {
             storage.addTopSite(title, url, isDefault)
+
+            scope.launch {
+                emitCountFact(storage.getTopSitesCount())
+            }
```
Comment (gabrielluong) [inline components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt:35]: How come we need a Coroutine? Isn't `emitCountFact ` synchronous?

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/facts/TopSitesFacts.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class TopSitesFacts {
+    /**
+     * Items that specify which portion of the [TopSitesFeature] was interacted with
+     */
+    object Items {
+        const val COUNT = "count"
+    }
+}
+
+internal fun emitCountFact(count: Int) {
```
Comment (gabrielluong) [inline components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/facts/TopSitesFacts.kt:24]:
```suggestion
internal fun emitTopSitesCountFact(count: Int) {
```

**Review by gabrielluong — APPROVED** [review]

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt (hunk 1 above)
Comment (rocketsroger) [reply components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt:35]: the `storage.getTopSitesCount()` is a database call so it should not be a blocking call.

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/facts/TopSitesFacts.kt (hunk 2 above)
Comment (rocketsroger) [reply components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/facts/TopSitesFacts.kt:24]: Sure, I'll rename.  Thanks,

File: components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt (hunk 1 above)
Comment (gabrielluong) [reply components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt]: gotcha

**Pull request merged**